lib/http/request.js

```diff
@@ -165,6 +169,10 @@ module.exports = (function() {
       self.emit('error', {}, response);
     });
 
+    this.request.setTimeout(this.timeoutDuration || 60000, function(){
```

Comment:
@diegoprd Agreed. I went ahead and implemented this for the timeout. in a latest commit